server/log_rpc_server_test.go

```diff
@@ -1231,38 +1234,91 @@ func TestGetConsistencyProofCommitFails(t *testing.T) {
 	test.executeCommitFailsTest(t, getConsistencyProofRequest7.LogId)
 }
 
-func TestGetConsistencyProof(t *testing.T) {
+// Ask for a proof from size 4 to 8 but the tree is only size 7. This should fail.
+func TestGetConsistencyProofOutsideTree(t *testing.T) {
 	ctrl := gomock.NewController(t)
 	defer ctrl.Finish()
 
 	mockStorage := storage.NewMockLogStorage(ctrl)
 	mockTx := storage.NewMockLogTreeTX(ctrl)
-	mockStorage.EXPECT().SnapshotForTree(gomock.Any(), getConsistencyProofRequest7.LogId).Return(mockTx, nil)
+	mockStorage.EXPECT().SnapshotForTree(gomock.Any(), getConsistencyProofRequest48.LogId).Return(mockTx, nil)
 
 	mockTx.EXPECT().LatestSignedLogRoot(gomock.Any()).Return(signedRoot1, nil)
-	mockTx.EXPECT().ReadRevision().Return(signedRoot1.TreeRevision)
-	mockTx.EXPECT().GetMerkleNodes(gomock.Any(), revision1, nodeIdsConsistencySize4ToSize7).Return([]storage.Node{{NodeID: stestonly.MustCreateNodeIDForTreeCoords(2, 1, 64), NodeRevision: 3, Hash: []byte("nodehash")}}, nil)
-	mockTx.EXPECT().Commit().Return(nil)
 	mockTx.EXPECT().Close().Return(nil)
 
 	registry := extension.Registry{
-		AdminStorage: mockAdminStorage(ctrl, getConsistencyProofRequest7.LogId),
+		AdminStorage: mockAdminStorage(ctrl, getConsistencyProofRequest48.LogId),
 		LogStorage:   mockStorage,
 	}
 	server := NewTrillianLogRPCServer(registry, fakeTimeSource)
 
-	response, err := server.GetConsistencyProof(context.Background(), &getConsistencyProofRequest7)
-	if err != nil {
-		t.Fatalf("failed to get consistency proof: %v", err)
+	_, err := server.GetConsistencyProof(context.Background(), &getConsistencyProofRequest48)
+	if err == nil {
+		t.Fatal("incorrectly returned a proof for out of range tree size")
 	}
+}
 
-	// Ensure we got the expected proof
-	expectedProof := trillian.Proof{
-		LeafIndex: 0,
-		Hashes:    [][]byte{[]byte("nodehash")},
+// Ask for a proof from size 5 to 4, testing the boundary condition. This request should fail
+// before making any storage requests.
+func TestGetConsistencyProofRangeError(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockStorage := storage.NewMockLogStorage(ctrl)
+	registry := extension.Registry{
+		AdminStorage: mockAdminStorage(ctrl, getConsistencyProofRequest54.LogId),
+		LogStorage:   mockStorage,
 	}
-	if !proto.Equal(response.Proof, &expectedProof) {
-		t.Fatalf("expected proof: %v but got: %v", expectedProof, response.Proof)
+	server := NewTrillianLogRPCServer(registry, fakeTimeSource)
+
+	_, err := server.GetConsistencyProof(context.Background(), &getConsistencyProofRequest54)
+	if err == nil {
+		t.Fatal("incorrectly returned a proof for out of range tree size")
+	}
+}
+
+func TestGetConsistencyProof(t *testing.T) {
```

Comment:
Seems like a lot of these tests could be collapsed into a table-driven one which includes a `wantError` flag, no?
Perhaps that could be done in a follow-up, though.

Reply:
Yes it could be done but seemed out of scope for this change.